src/python/zquantum/vqe/singlet_uccsd.py

```diff
@@ -0,0 +1,131 @@
+from zquantum.core.interfaces.ansatz import Ansatz
+from zquantum.core.interfaces.ansatz_utils import (
+    ansatz_property,
+    invalidates_parametrized_circuit,
+)
+from zquantum.core.circuit import Circuit
+from openfermion.utils import uccsd_singlet_paramsize, uccsd_singlet_generator
+from overrides import overrides
+from .utils import exponentiate_fermion_operator, create_layer_of_gates
+from pyquil import Program
+from pyquil.gates import X
+from typing import List, Optional
+import sympy
+import numpy as np
+
+
+class SingletUCCSDAnsatz(Ansatz):
+
+    supports_parametrized_circuits = False
+    transformation = ansatz_property("transformation")
+
+    def __init__(
+        self,
+        n_spatial_orbitals: int,
+        n_alpha: int,
+        n_layers: int,
+        transformation: str = "Jordan-Wigner",
+    ):
+        """ 
+        Ansatz class representing Singlet UCCSD Ansatz.
+
+        Args:
+            n_spatial_orbitals: number of spatial orbitals.
+            n_alpha: number of alpha electrons.
+            n_layers: number of layers of the ansatz.
+            transformation: transformation used for translation between fermions and qubits.
+
+        Attributes:
+            n_beta: number of beta electrons (equal to n_alpha).
```

Comment:
@jgonthier is `n_beta` always equal to `n_alpha`? If so should be bother even having this attribute?

Reply:
As @mstechly pointed out to me earlier, `n_alpha` is always equal to `n_beta` for the singlet ansatz. It's true that it may be a bit of a redundancy but having `n_beta` may allow to use some more general functions to build the ansatz.

Reply:
Yes, I thought that since they have different physical meaning it's good idea to keep the separate.
If we want to build other UCCSD ansatzes in the future, it'll be easier to handle this.

Reply:
Okay, that makes sense. Resolving conversation